yb-voyager/src/srcdb/yugabytedb.go

```diff
@@ -831,3 +834,88 @@ func (yb *YugabyteDB) GetNonPKTables() ([]string, error) {
 	}
 	return nonPKTables, nil
 }
+
+func (yb *YugabyteDB) GetReplicationConnection() (*pgconn.PgConn, error) {
+	return pgconn.Connect(context.Background(), yb.getConnectionUri()+"&replication=database")
+}
+
+func (yb *YugabyteDB) CreateLogicalReplicationSlot(conn *pgconn.PgConn, replicationSlotName string, dropIfAlreadyExists bool) (*pglogrepl.CreateReplicationSlotResult, error) {
+	if dropIfAlreadyExists {
+		log.Infof("dropping replication slot %s if already exists", replicationSlotName)
+		err := yb.DropLogicalReplicationSlot(conn, replicationSlotName)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	log.Infof("creating replication slot %s", replicationSlotName)
+	res, err := pglogrepl.CreateReplicationSlot(context.Background(), conn, replicationSlotName, "yboutput",
+		pglogrepl.CreateReplicationSlotOptions{Mode: pglogrepl.LogicalReplication})
+	if err != nil {
+		return nil, fmt.Errorf("create replication slot: %v", err)
+	}
+
+	return &res, nil
+}
+
+func (yb *YugabyteDB) DropLogicalReplicationSlot(conn *pgconn.PgConn, replicationSlotName string) error {
+	var err error
+	if conn == nil {
+		conn, err = yb.GetReplicationConnection()
+		if err != nil {
+			utils.ErrExit("failed to create replication connection for dropping replication slot: %s", err)
+		}
+		defer conn.Close(context.Background())
+	}
+	log.Infof("dropping replication slot: %s", replicationSlotName)
+
+	// TODO: Remove this sleep and check the status of the slot before dropping
+	// This sleep is added to avoid the error "replication slot is active" while dropping the slot since it takes 60 seconds by default to change the status of the slot
+	log.Info("sleeping for 60 seconds before dropping replication slot")
+	time.Sleep(60 * time.Second)
+
+	err = pglogrepl.DropReplicationSlot(context.Background(), conn, replicationSlotName, pglogrepl.DropReplicationSlotOptions{})
+	if err != nil {
+		// ignore "does not exist" error while dropping replication slot
+		if !strings.Contains(err.Error(), "does not exist") {
+			return fmt.Errorf("delete existing replication slot(%s): %v", replicationSlotName, err)
+		}
+	}
+	return nil
+}
+
+func (yb *YugabyteDB) CreatePublication(conn *pgconn.PgConn, publicationName string, tableList []sqlname.NameTuple, dropIfAlreadyExists bool, leafPartitions *utils.StructMap[sqlname.NameTuple, []string]) error {
+	if dropIfAlreadyExists {
+		err := yb.DropPublication(publicationName)
+		if err != nil {
+			return fmt.Errorf("drop publication: %v", err)
+		}
+	}
+	tablelistQualifiedQuoted := []string{}
+	for _, tableName := range tableList {
+		_, ok := leafPartitions.Get(tableName)
+		if ok {
+			//In case of partiitons, tablelist in CREATE PUBLICATION query should not have root
```

Comment:
Pls manually verify partitions behavior for YB in CREATE PUBLICATION. It may or may not be the same as PG.

Reply:
Okay